### PR TITLE
chore: loggers

### PR DESCRIPTION
**Pull request by saragee3**
{Includes} #8 

## Summary

`FileLogger` and `ConsoleLogger` to services/loggers. Ticket will be successfully closed when the other repos using `LoggerService` use `ConsoleLogger`. Am I missing anything else ?

## Other changes
- Bump Rhum version
- Minor fixes on prev tests
- Update README

### CODE REVIEW

File: loggers/file_logger.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+  /**
+   * Write a log message to this.file.
+   *
+   * @param message - The message to be logged
+   *
+   */
+  public write(message: string): void {
+    const encoder = new TextEncoder();
+    let encoded = encoder.encode(message + "\n");
```
Comment (ebebbington) [inline loggers/file_logger.ts]: Small nitpick... can this be a const?

Comment (saragee3) [reply loggers/file_logger.ts]: 8ca206d

**Comment by ebebbington** [comment]
@saragee3 i believe we need to add `Rhum.run()` at the end of the test files now we're using rhum v1

Along with this, i think we need to replace the running of the tests: `rhum-cli tests` --> `deno test -A tests` (as an example)

**Comment by saragee3** [comment]
I'm getting a leaking test. have you seen this error before: `AssertionError: Test case is leaking async ops.`

**Comment by crookse** [comment]
@saragee3, yea it's most likely due to an `async` function not being set up properly or the code is missing an `await` keyword somewhere

**Comment by ebebbington** [comment]
Isolate that teest case (commenting out every other test cases) and try see what might not be closing - causes of leaking ops are (i believe):

- not awaiting like eric said
- creating a process and not closing (eg websocket api, then not closing it) like:

```
Rhum.testcase(...), {
  const a = new WebSocket(...)
  // not closing mwuahaha
})

```

**Comment by saragee3** [comment]
yeahh odd removing the top level await works (bumper_service) but top level awaits are allowed in Deno...

**Comment by ebebbington** [comment]
https://github.com/drashland/services/pull/9/checks?check_run_id=1606621413:

image

Tells us a http request 'body' isn't being closed properly (i believe) - i can help debug it later

But afaik to stop leaking ops, fetch requests are closed by any of the following:

```typescript
await res.arrayBuffer() // for when you dont use/need the body or the text, purely to close the request
await res.json()
await res.text()
```

But it seems like very fetch req **internally** does at least one of those

**Comment by ebebbington** [comment]
i think it's the `await Rhum.testCase(...)` (the `await` bit) that's causing the problem?

**Comment by saragee3** [comment]
I've tried various async/await ops. I don't know how these were passing before... or if they were ever even in place. Will debug more next week -- I'll be on pto.

**Review by ebebbington — APPROVED** [review]
LGTM

**Review by crookse — COMMENTED** [review]
LGTM 🍻

**Comment by saragee3** [comment]
@crookse @ebebbington i'll release this under v0.1.0 sometime this week

**Review by crookse — APPROVED** [review]

**Comment by saragee3** [comment]
is this https://github.com/drashland/castle/blob/master/development_workflows.md#publishing-a-release accurate for this repo?

**Comment by ebebbington** [comment]
> is this https://github.com/drashland/castle/blob/master/development_workflows.md#publishing-a-release accurate for this repo?

Yeah, so you can merge this pr whenever you want, and to simplify the process, it's just:

1. When you want to release, create a `release-v<version to release>`branch
2. wait for and approve the pr that bumps the versions
3. update draft release
4. release

**Pull request merged**